packages/flutter_tools/lib/src/migrate/migrate_manifest.dart

```diff
@@ -0,0 +1,241 @@
+    final List<String> output = <String>[];
+    for (final String localPath in conflictFiles) {
+      if (_conflictsResolved(workingDir.childFile(localPath).readAsStringSync())) {
+        output.add(localPath);
+      }
+    }
+    return output;
+  }
+
+  /// A list of local paths of files that were automatically merged.
+  List<String> get mergedFiles {
+    final List<String> output = <String>[];
+    for (final MergeResult result in migrateResult.mergeResults) {
+      if (!result.hasConflict) {
+        output.add(result.localPath);
+      }
+    }
+    return output;
+  }
+
+  /// A list of local paths of files that were newly added.
+  List<String> get addedFiles {
+    final List<String> output = <String>[];
+    for (final FilePendingMigration file in migrateResult.addedFiles) {
+      output.add(file.localPath);
+    }
+    return output;
+  }
+
+  /// A list of local paths of files that are marked for deletion.
+  List<String> get deletedFiles {
+    final List<String> output = <String>[];
+    for (final FilePendingMigration file in migrateResult.deletedFiles) {
+      output.add(file.localPath);
+    }
+    return output;
+  }
+
+  /// Returns the manifest file given a migration workind directory.
+  static File getManifestFileFromDirectory(Directory workingDir) {
+    return workingDir.childFile('.migrate_manifest');
+  }
+
+  /// Writes the manifest yaml file in the working directory.
+  void writeFile() {
+    final StringBuffer mergedFileManifestContents = StringBuffer();
+    final StringBuffer conflictFilesManifestContents = StringBuffer();
+    for (final MergeResult result in migrateResult.mergeResults) {
+      if (result.hasConflict) {
+        conflictFilesManifestContents.write('  - ${result.localPath}\n');
+      } else {
+        mergedFileManifestContents.write('  - ${result.localPath}\n');
+      }
+    }
+
+    final StringBuffer newFileManifestContents = StringBuffer();
+    for (final String localPath in addedFiles) {
+      newFileManifestContents.write('  - $localPath\n)');
+    }
+
+    final StringBuffer deletedFileManifestContents = StringBuffer();
+    for (final String localPath in deletedFiles) {
+      deletedFileManifestContents.write('  - $localPath\n');
+    }
+
+    final String migrateManifestContents = 'merged_files:\n${mergedFileManifestContents.toString()}conflict_files:\n${conflictFilesManifestContents.toString()}added_files:\n${newFileManifestContents.toString()}deleted_files:\n${deletedFileManifestContents.toString()}';
+    final File migrateManifest = getManifestFileFromDirectory(migrateRootDir);
+    migrateManifest.createSync(recursive: true);
+    migrateManifest.writeAsStringSync(migrateManifestContents, flush: true);
+  }
+}
+
+/// Returns true if the file does not contain any git conflict markers.
+bool _conflictsResolved(String contents) {
+  if (contents.contains('>>>>>>>') && contents.contains('=======') && contents.contains('<<<<<<<')) {
```

Comment:
This should probably also fail if the markers have been partially removed. For example if the user removes the '>>>' and '===', but leaves in the '<<<'. Given that, I think this can be simplified as:

```dart
@visibleForTesting
bool conflictsResolved(String contents) {
  final bool hasMarker = contents.contains('>>>>>>>') ||
                         contents.contains('=======') ||
                         contents.contains('<<<<<<<');
  return !hasMarker;
}
```

This should also have a unit test.

Reply:
Ahh, he actually had it that way, and I asked him to use `&&`. But this makes sense, my bad.